### PR TITLE
added unitTests project, SignUpFormManager/IsEmailValid() test

### PR DESCRIPTION
**Pull request by juliusBaranauskas**
Pridedu IsEmailValid() testus

### CODE REVIEW

File: AppUnitTests/SignUpUnitTests.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+using System.Runtime.CompilerServices;
+using ServerCallFromApp;
+using ExternalDependencies;
+
+namespace AppUnitTests
+{
+    [TestClass]
+    public class SignUpUnitTests
+    {
+
```
Comment (Doombringer-MarAlith) [inline AppUnitTests/SignUpUnitTests.cs]: Unnecessary line

File: AppUnitTests/SignUpUnitTests.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        SignUpFormManager sign = new SignUpFormManager(new DataManipulations(new HttpInternalClient()), new FormManagerData());
```
Comment (Doombringer-MarAlith) [inline AppUnitTests/SignUpUnitTests.cs]: pervadinam `sign` I `formManager`

File: AppUnitTests/SignUpUnitTests.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        {
+            Assert.IsTrue(sign.IsEmailValid("x@example.com"));
+        }
+
+        [TestMethod]
+        public void TestIsEmailValid_UucpMail()
+        {
+            Assert.IsTrue(sign.IsEmailValid("mailhost!username@example.org"));
+        }
+
```
Comment (Doombringer-MarAlith) [inline AppUnitTests/SignUpUnitTests.cs]: Unnecessary line

File: AppUnitTests/SignUpUnitTests.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        {
+            Assert.IsFalse(sign.IsEmailValid("a\"b(c)d, e: f; g<h> i[j\\k]l @example.com"));
+        }
+
+        [TestMethod]
+        public void TestIsEmailValid_NoDotSeparator()
+        {
+            Assert.IsFalse(sign.IsEmailValid("just\"not\"right@example.com"));
+        }
+
```
Comment (Doombringer-MarAlith) [inline AppUnitTests/SignUpUnitTests.cs]: Unnecessary line

File: AppUnitTests/SignUpUnitTests.cs [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    [TestClass]
+    public class SignUpUnitTests
+    {
+
+        SignUpFormManager sign = new SignUpFormManager(new DataManipulations(new HttpInternalClient()), new FormManagerData());
+
+        [TestMethod]
+        public void TestIsEmailValid_EscapedRoute()
+        {
+            Assert.IsTrue(sign.IsEmailValid("user%example.com@example.org"));
```
Comment (BluePowerPlayer) [inline AppUnitTests/SignUpUnitTests.cs]: Yra toks dalykas kad nereikia rasyt kiekvienos eilutes i skirtinga metoda. Assertai gali buti multiple

File: AppUnitTests/SignUpUnitTests_Should.cs [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+using App;
+using Xunit;
+
+namespace AppUnitTests
+{
+	public class SignUpUnitTests_Should
+	{
+		[Theory]
+		[AutoMoqData]
+        public void TestIsEmailValid(SignUpFormManager sign)
```
Comment (BluePowerPlayer) [inline AppUnitTests/SignUpUnitTests_Should.cs]: Pervadinam i singUpFormManager

**Pull request merged**